src/main/java/seedu/rc4hdb/logic/commands/modelcommands/FilterCommand.java

```diff
@@ -0,0 +1,174 @@
+import seedu.rc4hdb.model.person.Name;
+import seedu.rc4hdb.model.person.Phone;
+import seedu.rc4hdb.model.tag.Tag;
+
+/**
+ * Filters and lists all persons in address book whose attributes are equal to any of the argument keywords.
+ * Keyword matching is case sensitive.
+ */
+public class FilterCommand extends ModelCommand {
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": filters the details of the person identified "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_PHONE + "PHONE] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com";
+
+    public static final String MESSAGE_FILTER_PERSON_SUCCESS = "filtered Person: %1$s";
+    public static final String MESSAGE_NOT_FILTERED = "At least one field to filter must be provided.";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+
+    private final FilterPersonDescriptor filterPersonDescriptor;
+
+    /**
+     * @param filterPersonDescriptor details to filter the person with
+     */
+    public FilterCommand(FilterCommand.FilterPersonDescriptor filterPersonDescriptor) {
+        requireNonNull(filterPersonDescriptor);
+        this.filterPersonDescriptor = new FilterCommand.FilterPersonDescriptor(filterPersonDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        //return new CommandResult("Command Still in progress");
+        requireNonNull(model);
+        AttributesMatchKeywordsPredicate predicate =
+                new AttributesMatchKeywordsPredicate(filterPersonDescriptor);
+        model.updateFilteredPersonList(predicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+    }
+
+    /**
+     * Stores the details to filter the person with. Each non-empty field value will replace the
+     * corresponding field value of the person.
+     */
+    public static class FilterPersonDescriptor {
+        private Name name;
+        private Phone phone;
+        private Email email;
+        private Address address;
+        private Set<Tag> tags;
+
+        public FilterPersonDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public FilterPersonDescriptor(FilterCommand.FilterPersonDescriptor toCopy) {
```

Comment:
I don't think we need to say that it's a `Copy constructor`. And do also consider using imperative mood, so something like `Constructs a ...`